### ISSUE TITLE
Cannot build ghcide on Windows

### ISSUE THREAD

**Issue by augustss** [haskell/ghcide#961]
The heapsize package refers to an undefined symbol.  Here's the build log:

Build log ( C:\Users\lennart.augustsson\AppData\Roaming\cabal\logs\ghc-8.10.3\ghcide-0.6.0.1-5nHKroEr0uE5lwJdL7Kohe.log ):
Preprocessing executable 'ghcide-test-preprocessor' for ghcide-0.6.0.1..
Building executable 'ghcide-test-preprocessor' for ghcide-0.6.0.1..
[1 of 1] Compiling Main             ( test\preprocessor\Main.hs, dist\build\ghcide-test-preprocessor\ghcide-test-preprocessor-tmp\Main.o )
Linking dist\build\ghcide-test-preprocessor\ghcide-test-preprocessor.exe ...
Preprocessing library for ghcide-0.6.0.1..
Building library for ghcide-0.6.0.1..
[ 1 of 52] Compiling Development.IDE.Compat ( src\Development\IDE\Compat.hs, dist\build\Development\IDE\Compat.o )
[ 2 of 52] Compiling Development.IDE.Core.Debouncer ( src\Development\IDE\Core\Debouncer.hs, dist\build\Development\IDE\Core\Debouncer.o )
[ 3 of 52] Compiling Development.IDE.Core.PositionMapping ( src\Development\IDE\Core\PositionMapping.hs, dist\build\Development\IDE\Core\PositionMapping.o )
[ 4 of 52] Compiling Development.IDE.GHC.Compat ( src\Development\IDE\GHC\Compat.hs, dist\build\Development\IDE\GHC\Compat.o )
[ 5 of 52] Compiling Development.IDE.GHC.CPP ( src\Development\IDE\GHC\CPP.hs, dist\build\Development\IDE\GHC\CPP.o )
[ 6 of 52] Compiling Development.IDE.Plugin.CodeAction.PositionIndexed ( src\Development\IDE\Plugin\CodeAction\PositionIndexed.hs, dist\build\Development\IDE\Plugin\CodeAction\PositionIndexed.o )
[ 7 of 52] Compiling Development.IDE.Session.VersionCheck ( session-loader\Development\IDE\Session\VersionCheck.hs, dist\build\Development\IDE\Session\VersionCheck.o )
ghc.exe:  | C:\Users\lennart.augustsson\AppData\Roaming\cabal\x86_64-windows-ghc-8.10.3\heapsize-0.3.0.1-LAg81MfrxlyBstfNO1PK7g\HSheapsize-0.3.0.1-LAg81MfrxlyBstfNO1PK7g.o: unknown symbol `heap_view_closurePtrs'
ghc.exe: Could not load Object Code C:\Users\lennart.augustsson\AppData\Roaming\cabal\x86_64-windows-ghc-8.10.3\heapsize-0.3.0.1-LAg81MfrxlyBstfNO1PK7g\HSheapsize-0.3.0.1-LAg81MfrxlyBstfNO1PK7g.o.

**Comment by jneira** [comment]
Hi, thanks for the bug report, are you using a commit that includes #954? that pr was supposed to fix that build error

**Comment by wz1000** [comment]
Perhaps we should make `heapsize` and optional dependency.

**Comment by jneira** [comment]
ops sorry, that pr only fixed the build for stack adding
```
configure-options:
    ghcide:
        - disable-library-for-ghci
    heapsize:
        - disable-library-for-ghci
```
but the cabal v2-install for windows and ghc-8.10.2 is not failing in ci, are you using cabal v1-install?

**Comment by jneira** [comment]
> 
> 
> Perhaps we should make `heapsize` and optional dependency.

Agree, hls is using the functionality added by heapsize, for example?

**Comment by pepeiborra** [comment]
I'm not interested in making `heapsize` an optional dependency just to work around an old bug in Cabal v1, but PRs are welcome. 

Lennart, the symbol is defined in the `ghc_heap` package. You just need to convince cabal v1 to not create the ghci libs, because those are not linked properly for some reason. Or just use either cabal v2 or stack

**Comment by augustss** [comment]
Yes, I'm using cabal v1, since I prefer that.
I'll see if I can turn off ghci libs, otherwise I'll just use my current
workaround which disables heapsize.

Thanks!

**Comment by jneira** [comment]
> I'm not interested in making heapsize an optional dependency just to work around an old bug in Cabal v1, but PRs are welcome.

agree in that it shouldn't be removed only for that, but if ghcide main usage is preserved without it, it introduces unnecessary dependencies downstream
